sycl/doc/extensions/SPIRV/SPV_INTEL_spec_conditional.asciidoc

```diff
@@ -0,0 +1,694 @@
+    specialization constant _Cond i_ associated with _Parent i_.
+--
++
+Validate that each `G(iuse)` can never be true unless `G(idef)` is also true.
+
+* For each block that is contained by the "then" range of
+  *OpBranchSpecConstantINTEL* or *OpBranchSpecConstantWithElseINTEL*, verify
+  that each parent block is also contained by that same "then" range (allowing,
+  of course, that the *OpBranchSpecConstantINTEL* or
+  *OpBranchSpecConstantWithElseINTEL* instruction is a parent of the first
+  block in that range).
+
+* For each block that is contained by the "else" range of
+  *OpBranchSpecConstantWithElseINTEL*, verify that each parent block is also
+  contained by that same "else" range (allowing, of course, that the
+  *OpBranchSpecConstantWithElseINTEL* instruction is a parent of the first
+  block in that range).
+
+* For each (_Condition i_, _Variable i_, _Parent i_) triplet of
+  *OpPhiSpecConstantINTEL*:
+  - If _Parent i_ resides in the "then" range of *OpBranchSpecConstantINTEL* or
+    *OpBranchSpecConstantWithElseINTEL*, verify that _Condition i_ is the same
+    specialization constant _<id>_ as the *OpBranchSpecConstantINTEL* or
+    *OpBranchSpecConstantWithElseINTEL* instruction.
+
+  - If _Parent i_ is the _False Label_ of *OpBranchSpecConstantINTEL* or
+    _Parent i_ resides in the "else" range of
+    *OpBranchSpecConstantWithElseINTEL*, verify that _Condition i_ is the
+    _<id>_ of a specialization constant that is the logical negation of the
+    specialization constant _<id>_ used by the *OpBranchSpecConstantINTEL* or
+    *OpBranchSpecConstantWithElseINTEL* instruction.
+
+  - Otherwise, verify that _Condition i_ has the value zero.
+
+  - If _Variable i_ is defined by a block in the "then" range of
+    *OpBranchSpecConstantINTEL* or *OpBranchSpecConstantWithElseINTEL*, or if
+    _Variable i_ is defined by a block in the "else" range of
+    *OpBranchSpecConstantWithElseINTEL*, verify that _Parent i_ is contained by
+    that same "then" or "else" range.
+
+* For each specialization constant (*OpSpecConstantXXX*) that is decorated with
+  *ConditionalINTEL*, verify that the specialization constant's _<id>_ is not
+  also used as a _Condition_ for *OpConditionalCapabilityINTEL*,
+  *OpBranchSpecConstantINTEL*, *OpBranchSpecConstantWithElseINTEL*,
+  *OpPhiSpecConstantINTEL*, or *ConditionalINTEL*.
+
+*TODO*: I'm looking for feedback on the validation rules that involve
+comparison of abstract specialization constants such as "validate that `G(i)`
+can never be true unless `G(c)` is also true".  These validations are likely
+very difficult in the general case, but most cases will be very easy.  (In most
+cases, I think that `G(i)` and `G(c)` will both be so simple that it will be
+easy to prove whether one implies the other.)  How should the spec handle this?
+Are the validation rules _requirements_ that a validation tool must perform, or
+are they just rules about what SPIR-V is not valid?  If they are not
```

Comment:
SPIR-V spec doesn't say anything about validation tool and its requirements, but from point of view of a developer of such tool, I would implement all available validation rules to provide as precise result as possible.

To me it seems that validation rules are made for SPIR-V generators so they prepare SPIR-V in a particular way, which is expected (and assumed, covered by `assert`s) in generators. Consumer is free to avoid performing any validation at all.

Reply:
Resolving this because we decided to go with an alternate design that does not require a SPIR-V extension.